src/main/scala/firrtl/FirrtlExecutionOptions.scala

```diff
@@ -0,0 +1,752 @@
+final case class OneFilePerModule(targetDir: String) extends OutputConfig
+
+/** Internal options used to control the FIRRTL compiler
+  *
+  * @param topName the name of the top module
+  * @param targetDirName name of the target directory (default: ".")
+  * @param globalLogLevel the verbosity of logging (default: [[logger.LogLevel.None]])
+  * @param classLogLevels the individual verbosity of logging for specific classes
+  * @param logToFile if true, log to a file
+  * @param logClassNames indicates logging verbosity on a class-by-class basis
+  * @param programArgs explicit program arguments
+  * @param inputFileNameOverride input FIRRTL file, default: `targetDir/topName.fir`
+  * @param outputFileNameOverride output file, default: `targetDir/topName.SUFFIX` with `SUFFIX`
+  *                               as determined by the compiler
+  * @param outputAnnotationFileName the output annotation filename, default: `targetDir/topName.anno`
+  * @param compilerName which compiler to use (default: "verilog")
+  * @param infoModeName the policy for generating [[firrtl.ir.Info]] when processing FIRRTL (default: "append")
+  * @param customTransforms any custom [[Transform]] to run
+  * @param firrtlSource explicit input FIRRTL as a [[scala.String]]
+  * @param annotations a sequence of [[annotations.Annotation]] passed to the compiler
+  * @param emitOneFilePerModule enables one-file-per-module output in the [[Emitter]]
+  */
+final case class FirrtlExecutionOptions(
+  topName:                  Option[String]              = None,
+  targetDirName:            String                      = ".",
+  globalLogLevel:           LogLevel.Value              = LogLevel.None,
+  classLogLevels:           Map[String, LogLevel.Value] = Map.empty,
+  logToFile:                Boolean                     = false,
+  logClassNames:            Boolean                     = false,
+  programArgs:              Seq[String]                 = Seq.empty,
+  inputFileNameOverride:    Option[String]              = None,
+  outputFileNameOverride:   Option[String]              = None,
+  outputAnnotationFileName: Option[String]              = None,
+  compilerName:             String                      = "verilog",
+  infoModeName:             String                      = "append",
+  customTransforms:         Seq[Transform]              = List.empty,
+  firrtlSource:             Option[String]              = None,
+  annotations:              List[Annotation]            = List.empty,
+  emitOneFilePerModule:     Boolean                     = false) {
+
+  /** Return the info mode */
+  def infoMode(): Parser.InfoMode = {
+    infoModeName match {
+      case "use"    => Parser.UseInfo
+      case "ignore" => Parser.IgnoreInfo
+      case "gen"    => Parser.GenInfo(inputFileNameOverride.getOrElse("[not defined]"))
+      case "append" => Parser.AppendInfo(inputFileNameOverride.getOrElse("[not defined]"))
+      case _        => Parser.UseInfo
```

Comment:
Was this like this before? We don't want people to be able to give "boijwfoij" and just get `UseInfo` right?

Reply:
Apparently it was! However, there was a validation check during option parsing to make sure that `infoModeName` was one of the allowable strings. That same check is still in place, however, if the user passes in this option via an annotation file, then that check won't happen. This needs to be fixed, or the check needs to be moved somewhere else.